src/middleware/cache.rs

```diff
@@ -0,0 +1,56 @@
+use async_trait::async_trait;
+use jsonrpsee::core::{Error, JsonValue};
+use schnellru::{ByLength, LruMap};
+use std::sync::Mutex;
+
+use super::{Middleware, NextFn, Request};
+
+pub struct CacheMiddleware {
+    whitelist: Vec<String>,
+    map: Mutex<LruMap<(String, Vec<String>), JsonValue>>,
```

Comment:
this will kill the performance.

should be something like `DashMap<String, RwLock<LruMap<String, String>>>` so that methods don't share lock

Reply:
now it's method has it's own cache

Reply:
we can't use one mutex for the whole cache